src/linelast_solver.cpp

```diff
@@ -438,22 +453,30 @@ void LinElastSolver::SetParameterizedProblem(ParameterizedProblem *problem)
       mu_c[i] = new ConstantCoefficient(mu_i);
    }
 
-   // Set BCs
-   for (int b = 0; b < problem->battr.Size(); b++)
+   // Set BCs, the switch on BC type is done inside SetupRHSBCOperators
+   for (int b = 0; b < global_bdr_attributes.Size(); b++)
    {
-      switch (problem->bdr_type[b])
+      int ti = problem->battr.Find(global_bdr_attributes[b]);
+
+      if (ti >=0)
+      {
+      type_idx[b] = (LinElastProblem::BoundaryType)problem->bdr_type[ti];
+      switch (problem->bdr_type[ti])
```

Comment:
Do we need this `switch`? They seem to have the same behavior.

Reply:
Probably not